[hotfix][doc] add content for using numRecordsSend and deprecating numRecordsOutErrors (#19664)
apache/flink: docs/content/release-notes/flink-1.15.md

The old JDBC connector (indicated by `connector.type=jdbc` in DDL) has been removed.
If not done already, users need to upgrade to the newer stack (indicated by `connector=jdbc` in DDL).

> #### [FLINK-26126](https://issues.apache.org/jira/browse/FLINK-26126)

Reviewer: I think it would be good to add a separate headline as "Extensible unified Sink uses new metric to capture outgoing records".

Author: Excellent! I forgot that.

New metric `numRecordsSend` has been introduced to denote the number of records sent to the external system 

> while the numRecordsOut will be used to denote the number of records transferred between tasks.

Reviewer: ```suggestion
while the numRecordsOut will be used to denote the number of records transferred between sink tasks.
```

> Considering naming consistency, `numRecordsOutErrors` is deprecated, please use `numRecordsSendErrors` instead.

Reviewer: WDYT about adding two different paragraphs? The first describes the perspective of a connector user that now needs to use `numRecordsSend` `numRecordsSendErrors` to monitor the records going to the external system.
The second paragraph holds information for connector developers that have to eventually update their connectors to use the new metrics because `numRecordsOurErrors` is deprecated and `numRecordsOut` has a different meaning.

The old JDBC connector (indicated by `connector.type=jdbc` in DDL) has been removed.
If not done already, users need to upgrade to the newer stack (indicated by `connector=jdbc` in DDL).

#### Extensible unified Sink uses new metric to capture outgoing records

##### [FLINK-26126](https://issues.apache.org/jira/browse/FLINK-26126)

> New metrics `numRecordsSend` and `numRecordsSendErrors` has been introduced for users to monitor the number of 

Reviewer: ```suggestion
New metrics `numRecordsSend` and `numRecordsSendErrors` have been introduced for users to monitor the number of 
```

records sent to the external system. The `numRecordsOut` should be used to monitor the number of records 
transferred between sink tasks.

> Connector developers should Consider using the new metric `numRecordsSendErrors` to monitor errors, 

Reviewer: ```suggestion
Connector developers should consider using the new metric `numRecordsSendErrors` to register errors when sending to the external system instead of `numRecordsOutErrors` which is deprecated from now on.
```

> since `numRecordsOut` has a different and more general meaning, and `numRecordsOutErrors` is therefore deprecated.

Reviewer: ```suggestion
Additionally since `numRecordsOut` now also counts the records sent between sink tasks we recommend changing the connector to use `numRecordsSend` to be fully accurate.
```

Author: This is confused for users and connector developers, tbh.  1. `numRecordsOut` will ONLY count the records(technically Commitables) sent between sink tasks. 2. the information about the deprecated metric `numRecordsOutErrors` is missing. 3. the information of using `numRecordsSend` has been described in the first part and there is no use case for "changing the connector"(all connectors that used `numRecordsOut` in a wrong way have been changed). From now on, it is actually mandatory for connector developers to use them except `numRecordsOutErrors`. 

How about this version:

Connector developers should pay attention to the usage these metrics `numRecordsOut`, `numRecordsSend` and `numRecordsSendErrors` while building your sink connectors. Please refer to the new Kafka Sink for more information. Additionally since `numRecordsOut` now only counts the records sent between sink tasks and `numRecordsOutErrors` was designed for counting the records sent to the external system, we deprecated `numRecordsOutErrors` and recommend using `numRecordsSendErrors` instead.

Event with this description, it is still unclear, since `numRecordsOut` is still used, why is `numRecordsOutErrors` deprecated? It should be used to monitor errors happened while Commitables were send between sink tasks. WDYT?

Reviewer: Sounds good to me 👍
